### DISCUSSION TITLE
netbox

### DISCUSSION THREAD

**Discussion by kc0hwa** [netbox-community/netbox#16691] [General]
can netbox do generic setup before buying hardware
then when you find the hardware you add the hardware

**Comment by candlerb** [comment]
It depends what you mean by "generic setup".

If you mean allocating IP addresses, then the IPAM works fine without any devices. If you mean creating sites and racks, then that's fine (you'd need to do that before adding devices anyway).

Or do you mean you want to create rack layouts, before you know exactly what type of devices you're getting?

You can change the device type of a device after it has been created. So for example you could create dummy devices for "generic 1u server", "generic 2u server" etc, and then replace them later.

However, if you do change the device type on an existing device, it will not gain any of the templated components (e.g. interfaces, power ports) which are in the device type.

There's a script you can use to copy these across from the device type:
https://github.com/netbox-community/customizations/blob/master/scripts/add_device_type_components.py

although there are a couple of outstanding pull requests which haven't been applied:
* https://github.com/netbox-community/customizations/pull/101
* https://github.com/netbox-community/customizations/pull/107
